src/components/BarChart/index.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from "react";
+import { Bar } from "react-chartjs-2";
+import { generateRandomColors, createColorsWithOpacity } from "../../shared/utils";
+
+const CHART_COUNT = 8;
+
+const BarChart = ({ data }) => {
+    const colors = generateRandomColors(CHART_COUNT);
+    const colorsOpacity = createColorsWithOpacity(colors);
+    const [backgroundColor, setBackgroundColor] = useState([]);
+    const [prevBackgroundColor, setPrevBackgroundColor] = useState();
+    const [colorsWithOpacity, setColorsWithOpacity] = useState([]);
+    const [prevColorsWithOpacity, setPrevColorsWithOpacity] = useState();
+    const [state, setState] = useState();
+    const [options, setOptions] = useState();
+    const [index, setIndex] = useState();
+
+    useEffect(() => {
+        if (backgroundColor.length !== 8) {
```

Comment:
noob checking since if youll have 9 colors there this will stop working

Reply:
done